### PR TITLE
[fix] correct headers for __data.js requests

### LINKED ISSUE #6458: Cache problem with client-side navigation: the response is now __data.js instead of __data.json
### Describe the bug

I believe the recent change where client-side navigation responds with __data.js instead of __data.json can be problematic.

I have Cloudflare on my site and I just realized that the __data.js files that are returned when navigating between pages are being cached on the edge and possibly also in the browser cache. I don't have any custom page rules for .js files in Cloudflare, this is the default behavior.

In some cases, the "age" in the response headers is showing very high numbers, much higher than what I have in my s-max-age cache-control headers in hooks.js (which used to cache on Vercel edge only, not Cloudflare).

681e4b3be00bfa462433bc8450004e6b

This means that many users are potentially seeing very old and outdated info for pages that are supposed to be highly dynamic.

### Reproduction

I don't think I can set up a reproduction since it would require a custom domain and adding it to Cloudflare.

### Logs

_No response_

### System Info

```shell
System:
    OS: Windows 10 10.0.22000
    CPU: (32) x64 AMD Ryzen 9 5950X 16-Core Processor
    Memory: 50.08 GB / 63.91 GB
  Binaries:
    Node: 16.16.0 - C:\Program Files\nodejs\node.EXE
    npm: 8.11.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.22000.120.0), Chromium (104.0.1293.70)
    Internet Explorer: 11.0.22000.120
  npmPackages:
    @sveltejs/adapter-auto: 1.0.0-next.69 => 1.0.0-next.70
    @sveltejs/kit: 1.0.0-next.454 => 1.0.0-next.456
    svelte: ^3.49.0 => 3.49.0
    vite: 3.1.0-beta.1 => 3.1.0-beta.1
```

### Severity

Blocking an upgrade

### Additional Information

Update: I rolled back to 446 for now since the 447 update seems to also have introduced a bunch of weird issues with both Sentry and Plausible Analytics.

### PR DESCRIPTION
**Pull request by dummdidumm**
Add all headers except cache-control which is always no-store
Fixes #6458
Fixes (?) #6477 <- not sure how to best handle this, conflicting requirements here

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [x] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. All changesets should be `patch` until SvelteKit 1.0

### CODE REVIEW

File: .changeset/cyan-humans-perform.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@sveltejs/kit': patch
+---
+
+[fix] correct headers for \_\_data.js requests
```
Comment (Rich-Harris) [inline .changeset/cyan-humans-perform.md:5]:
```suggestion
[fix] correct headers for `__data.js` requests
```

**Comment by Rich-Harris** [comment]
I'm not sure we want to add headers other than `set-cookie`? There's no way for the application to access them so they're useless at best (except `set-cookie` and `cache-control`, which have side-effects). 

Given that some platforms sometimes apparently apply default `cache-control` headers if none are provided, and given that the `__id` parameter is per-user, I think we need to always apply `private, no-store`. If we respect `max-age` and `s-maxage` it should probably be in the form of a `window.__sveltekit_data_ttl` value or similar, the same way we cache data inlined into the HTML response.

**Comment by dummdidumm** [comment]
I'm not well-versed with all the possibilities of headers, but in  #6477 on comment asked for setting headers besides the cache-control - don't know if that's sensible.

**Comment by homerjam** [comment]
> not rerunning them is ensured differently
So is there another way to ensure data requests are not re-run unnecessarily? Is `window.__sveltekit_data_ttl` documented somewhere?

**Comment by Rich-Harris** [comment]
> on comment asked for setting headers besides the cache-control - don't know if that's sensible

Those `sec-fetch-mode` and `sec-fetch-dest` headers would have no effect

> Is `window.__sveltekit_data_ttl` documented somewhere?

No, it doesn't exist. If we do implement it, it'll be an internal detail, not something you'd need to be aware of

**Comment by dummdidumm** [comment]
Closing as obsolete - the no-store part was implemented in another PR and the cache will need a different implementation.

**Pull request closed without merging**